schematic/store/synapse.py

```diff
@@ -51,9 +51,98 @@
 from schematic.exceptions import MissingConfigValueError, AccessCredentialsError
 
 from schematic import CONFIG
+from schematic.utils.general import entity_type_checking
 
 logger = logging.getLogger("Synapse storage")
 
+class ManifestDownload(object):
+    def _download_manifest_to_folder(self, syn, manifest_id):
+        """
+        try downloading a manifest to local cache or a given folder
+        manifest
+        Args:
+            syn: an object of type synapseclient.
+            manifest_id: id of a manifest  
+        Return: 
+            manifest_data: A new Synapse Entity object of the appropriate type
+        """
+        if CONFIG["synapse"]["manifest_folder"]:
+            manifest_data = syn.get(
+                    manifest_id,
+                    downloadLocation=CONFIG["synapse"]["manifest_folder"],
+                    ifcollision="overwrite.local",
+                )
+        else:
+            manifest_data = syn.get(
+                        manifest_id,
+                    )
```

Comment:
Makes sense, I'll relay what I said in slack, which is to slowly lint each script with `black` and look through the code changes.  Personally, I've never encountered any code breaking changes with `black`

Reply:
hmm @thomasyu888 I tried linting `synapse.py` but since that file is huge, it looks like there are a lot of formatting changes. Should I create a separate PR to lint the module after this one gets merged? Since otherwise it would be difficult to see the changes that I made to the module. `general.py` is much shorter so I linted that in this PR.

Reply:
So, I would try to do linting of entire modules completely separate (general.py is probably fine).  I know there is actually a vscode module that automatically lints only the code you touch. I'd look into that.

There are also auto docstrings creators in vscode, If you have type hints in your function, it will create a nice docstrings template for you to fill out.